### PR TITLE
Clean assets and target after test run

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Run Maven clean after running tests to clean up target folders and deletes .assets.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]

File: eng/pipelines/templates/steps/clean-maven-local-cache.yml [hunk 1]
```diff
@@ -7,3 +7,9 @@ steps:
       arguments: >
         -MavenCacheFolder $(MAVEN_CACHE_FOLDER)
     continueOnError: true
+
+  - pwsh: |
```
Comment (benbp) [inline eng/pipelines/templates/steps/clean-maven-local-cache.yml:11]: Should we just change the name of this template to be a bit clearer and move all the above steps you added into it too?

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 2]
```diff
@@ -148,6 +148,32 @@ jobs:
           TestParallelization: ${{ parameters.TestParallelization }}
           BuildParallelization: ${{ parameters.BuildParallelization }}
 
+      - task: Maven@3
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply eng/pipelines/templates/jobs/ci.tests.yml:151]: It is, will add a comment about that

File: eng/pipelines/templates/steps/clean-maven-local-cache.yml (hunk 1 above)
Comment (alzimmermsft) [reply eng/pipelines/templates/steps/clean-maven-local-cache.yml:11]: Yeah, I think we can call this `post-job-maven-clean`, or something like that

**Review by benbp — APPROVED** [review]

**Pull request merged**